Add mode to ddg2dot.py to make differences in edges apparent

Quincunx271 opened the pull request:
When edges are added or removed via graph transformations, comparing the
DDGs via ddg2dot.py would be rather difficult. This augments ddg2dot.py
to add invisible edges to encourage `dot` to produce the graphs in the
same layout, with just the visibility of the edges being different.

<details>
<summary>Example graphs augmented with this new feature.</summary>

![kernel_c101_sdk_900 18 ddg](https://user-images.githubusercontent.com/7417022/105248125-b8ac8480-5b2a-11eb-929e-3f4f8472a709.png)

![kernel_c101_sdk_900 18 occupancy-preserving-ilp nodesup ddg](https://user-images.githubusercontent.com/7417022/105248134-bc400b80-5b2a-11eb-82b9-d96739ec2091.png)
</details>

paul-mchugh commented on util/misc/ddg2dot.py:
```diff
@@ -30,37 +44,71 @@
 output = sys.stdout
-if args.output: output = open(args.output, 'w')
+if args.output:
+    output = open(args.output, 'w')
```
The output file descriptor is not closed by the time the program ends.  This is a nit and the file will be closed anyway when the program exits, so I am okay with not doing anything about this comment.  But resource leaks are usually considered bad form.

paul-mchugh reviewed (commented):
I haven't used your ddg2dot.py script, so any changes to this script will not effect my workflows. I have a couple nits and questions.

Why did you split some lines into two lines?  It is my understanding that the 80 character rule only applies to c++.  Where it doesn't apply, I don't think we should split up the lines since I think it makes it harder to read(on larger screens)(I dont think any of us view our code on an 80 char display). eg: see lines 8-15.

My other nit is the is the inline comment on line 112.

Quincunx271 commented:
>Why did you split some lines into two lines?

I now have a Python code formatter (autopep8) installed on my machine, so I applied it to the file. I didn't mess with any of the formatter's settings, and PEP8 prescribes an 80 character line limit, so it wrapped the lines. I tend to agree with you that it's not necessary and somewhat hampers readability, but I'd prefer to leave it formatted according to autopep8 unless we figure some configuration file like that .clang-format file.

Quincunx271 commented:
@paul-mchugh I addressed your comments. I found how to make a configuration file for autopep8 as well, so the unnecessary line-wrapping is reverted now.

paul-mchugh reviewed (approved):
LGTM
